Algorithm.CSharp/CustomDataUnderlyingSymbolMappingRegressionAlgorithm.cs

```diff
@@ -0,0 +1,190 @@
+                    var symbol = data.SymbolChangedEvents.Keys.Where(x => x.SecurityType == SecurityType.Equity && x == _equitySymbol).Single();
+
+                    if (SubscriptionManager.Subscriptions.Where(x => (x.SecurityType == SecurityType.Base || x.SecurityType == SecurityType.Equity) && x.MappedSymbol == expectedUnderlying).Count() != 2)
+                    {
+                        throw new Exception($"Subscription mapped symbols were not updated to {expectedUnderlying}");
+                    }
+                    if (underlying == null)
+                    {
+                        throw new Exception("Custom data Symbol for GOOGL has no underlying");
+                    }
+                    if (underlying != symbol)
+                    {
+                        throw new Exception($"Underlying custom data Symbol does not match equity Symbol after rename event. Expected {symbol.Value} - got {underlying.Value}");
+                    }
+                    if (underlying.Value != expectedUnderlying)
+                    {
+                        throw new Exception($"Underlying equity symbol value from chained custom data does not match expected value. Expected {symbol.Underlying.Value}, found {underlying.Underlying.Value}");
+                    }
+
+                    SetHoldings(symbol, 0.5);
+                }
+                else if (data.SymbolChangedEvents.ContainsKey(_badCustomDataSymbol) && data.SymbolChangedEvents.ContainsKey(_badEquitySymbol))
+                {
+                    var underlying = data.SymbolChangedEvents.Keys.Where(x => x.SecurityType == SecurityType.Base && x == _badCustomDataSymbol).Single().Underlying;
+                    var symbol = data.SymbolChangedEvents.Keys.Where(x => x.SecurityType == SecurityType.Equity && x == _badEquitySymbol).Single();
+
+                    if (underlying == null)
+                    {
+                        throw new Exception($"Bad custom data symbol does not have underlying");
+                    }
+                    if (underlying == symbol)
+                    {
+                        throw new Exception($"Underlying custom data Symbol is equal to bad Symbol");
+                    }
+                }
+                else
+                {
+                    throw new Exception("Received unknown symbol changed event");
+                }
+            }
+        }
+
+        public override void OnSecuritiesChanged(SecurityChanges changes)
+        {
+            foreach (var added in changes.AddedSecurities.Where(x => x.Symbol.SecurityType == SecurityType.Equity))
+            {
+                // It is in fact "GOOGL" we're catching here, and we're adding it as "GOOG" with the ticker,
+                // which will resolve to GOOCV in the past if we use the ticker and not the symbol
+                if (added.Symbol.Value == "GOOG")
```

Comment:
Do we need this `if` statement, why? If possible, think it would be nice if this regression algorithm showed cased the final user use case